File: src/test/java/android/util/AndroidSDKResolver.java
package android.util;

import io.appium.java_client.android.AndroidDriver;

import java.util.HashMap;
import java.util.Map;

public final class AndroidSDKResolver {

    private static final Map<String, Integer> sdkMap = new HashMap<String, Integer>() {{
        put("5.0", 21);
        put("5.1", 22);
        put("6.0", 23);
        put("7.0", 24);
        put("7.1", 25);
        put("8.0", 26);
        put("8.1", 27);
        put("9", 28);
        put("10", 29);
        put("11", 30);
        put("12", 31);
    }};

    public static int resolve(final AndroidDriver driver) {
        final String platformName = (String) driver.getCapabilities().getCapability("platformVersion");
        for (final Map.Entry<String, Integer> entry : sdkMap.entrySet()) {
            if (platformName.startsWith(entry.getKey())) {
                return entry.getValue();
            }
        }

        System.out.println("Android version not recognised: " + platformName);
        return 1;
    }
}

File: src/test/java/android/util/AndroidSDKResolver.kt
package android.util

import io.appium.java_client.android.AndroidDriver

object AndroidSDKResolver {

    private val sdkMap: Map<String, Int> = mapOf(
        "5.0" to 21,
        "5.1" to 22,
        "6.0" to 23,
        "7.0" to 24,
        "7.1" to 25,
        "8.0" to 26,
        "8.1" to 27,
        "9" to 28,
        "10" to 29,
        "11" to 30,
        "12" to 31,
        "12L" to 32,
        "13" to 33,
        "14" to 34,
        "15" to 35,
        "16" to 36
    )

    fun resolve(driver: AndroidDriver): Int {
        val platformName = driver.capabilities.getCapability("platformVersion") as String
        for ((key, value) in sdkMap) {
            if (platformName.startsWith(key)) {
                return value
            }
        }
        println("Android version not recognised: $platformName")
        return 1
    }
}